akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -388,4 +405,60 @@ private[akka] class PersistentShard(
     }
   }
 
+  private def retrieveEntityRecoveryConfig(): Config = {
+    val path = settings.tuningParameters.entityRecoveryStrategyConfigPath
+    if (path.isEmpty)
+      ConfigFactory.empty()
+    else
+      context.system.settings.config.getConfig(path)
+  }
+}
+
+abstract class EntityRecoveryStrategy {
+  import ShardRegion.EntityId
+  import akka.actor.ActorContext
+  def recoverEntities(context: ActorContext, entities: Set[EntityId])
```

Comment:
Lucky him! :-) 

Just we are on the same page here... There seems to be two possible solutions:
1.) Provide 2 fixed implementations and implement them in the shard actor with config that allows us to switch between them. That possibly being 

`entity-recovery-strategy = "<all|constant>"` 

2.) Keep the current implemented interfaces; apply Patrik suggestions (ActorContext and the API return a Future); remove the external pluggable mechanism in favour config stated in 1.

I think you are siding with option 2, ya?